extensions/app-search/config/elasticsearch.yml

```diff
@@ -0,0 +1,31 @@
+---
+## Default Elasticsearch configuration from Elasticsearch base image.
+## https://github.com/elastic/elasticsearch/blob/master/distribution/docker/src/docker/config/elasticsearch.yml
+#
+cluster.name: "docker-cluster"
+network.host: 0.0.0.0
+
+## X-Pack settings
+## see https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-xpack.html
+#
+## Configs without app-search
+# xpack.license.self_generated.type: trial
+# xpack.monitoring.collection.enabled: true
+# xpack.security.enabled: true
+
+## Configs with app-search
+action.auto_create_index: ".app-search-*-logs-*,-.app-search-*,+*"
+
+# These options were used get app-search working
+# App search requires you to use a basic license
+# xpack.license.self_generated.type: trial
+xpack.license.self_generated.type: basic
+xpack.monitoring.collection.enabled: true
+xpack.security.enabled: true
+xpack:
+    security:
+      authc:
+        realms:
+          native:
+            native1:
+              order: 0
```

Comment:
Since the required configuration changes are moderate (license, realm and create_index if I understood correctly?), I would prefer to see them documented in the README instead of in a separate file.

Two reasons for that:

* Users might want to run other extensions in parallel, in which case we encourage them to consciously do the changes required for their own needs.
* If we add or change some of our defaults in the future, we will also have to juggle between multiple config files and perform the changes in multiple locations.

A code snippet in your README should be enough.

Example:

> This extension requires you to perform the following changes to the Elasticsearch configuration file:
> * Use the basic license
> * Allow the automation creation of App Search's indexes
> * Configure the authentication realms

```yaml
action.auto_create_index: ". app-search-*-logs-*,-.app-search-*,+*"

xpack.license.self_generated.type: basic

xpack:
  security:
    # ...
```

Reply:
Apparently these settings are not even required for App Search to work (see my comment in the main thread)

Reply:
I put these in here so you'd have an app_search user in a consistent experience to the other apps, but have removed them.